packages/compass-e2e-tests/helpers/commands/connect.ts

```diff
@@ -0,0 +1,233 @@
+    // if the modal is still animating away when we're connecting again, things
+    // are going to get confused
+    await browser
+      .$(Selectors.ConnectionModal)
+      .waitForDisplayed({ reverse: true });
+
+    // if a connection with this name already exists, remove it otherwise we'll
+    // add a duplicate and things will get complicated fast
+    const connectionName = connectionNameFromString(connectionString);
+    if (await browser.removeConnection(connectionName)) {
+      debug('Removing existing connection so we do not create a duplicate', {
+        connectionName,
+      });
+    }
+
+    await browser.clickVisible(Selectors.Multiple.SidebarNewConnectionButton);
+    await browser.$(Selectors.ConnectionModal).waitForDisplayed();
+  }
+
+  await browser.setValueVisible(
+    Selectors.ConnectionFormStringInput,
+    connectionString
+  );
+
+  const connectionName = connectionNameFromString(connectionString);
+  await browser.doConnect(connectionName, options);
+}
+
+export async function connectWithConnectionForm(
+  browser: CompassBrowser,
+  state: ConnectFormState,
+  options: ConnectOptions = {}
+): Promise<void> {
+  // Use this command when you need to add a new connection with specific
+  // connect form field values. Most test files should just be using
+  // browser.connectToDefaults()
+
+  // If a connectionName is specified and a connection already exists with this
+  // name, make sure we don't add a duplicate so that tests can always address
+  // this new connection.
+  if (state.connectionName) {
+    if (await browser.removeConnection(state.connectionName)) {
+      debug('Removing existing connection so we do not create a duplicate', {
+        connectionName: state.connectionName,
+      });
+    }
+  }
+
+  await browser.setConnectFormState(state);
+
+  if (!state.connectionName) {
+    // In theory we could calculate the auto-generated connectionName here or
+    // try an read it out.
+    throw new Error('state.connectionName is required');
+  }
+  const connectionName = state.connectionName;
+  await browser.doConnect(connectionName, options);
+}
+
+export async function doConnect(
+  browser: CompassBrowser,
+  connectionName: string,
+  options: ConnectionResultOptions = {}
+) {
+  await browser.clickVisible(Selectors.ConnectButton);
+  await browser.waitForConnectionResult(connectionName, options);
+}
+
+export async function waitForConnectionResult(
+  browser: CompassBrowser,
+  connectionName: string,
+  { connectionStatus = 'success', timeout }: ConnectionResultOptions = {}
+) {
+  const waitOptions = typeof timeout !== 'undefined' ? { timeout } : undefined;
+
+  if (connectionStatus === 'either') {
+    // TODO(COMPASS-7600): this doesn't support compass-web yet, but also isn't
+    // encountered yet For the rare cases where we don't care whether it fails
+    // or succeeds
+    await browser
+      .$(`${Selectors.DatabasesTable},${Selectors.ConnectionFormErrorMessage}`)
```

Comment:
the error part will not work for multiple connections because it's missing the toast click

Reply:
This is also old code I moved. It has probably just never been encountered. The next subtask is to test the multiple connection errors and then I'll have to change this. There's only one test that uses this and it is currently skipped for multiple connections.

Reply:
I updated the TODO comment.